src/sdkg.rs

```diff
@@ -339,20 +344,20 @@ impl<N: NodeIdT> SyncKeyGen<N> {
     pub fn count_complete(&self) -> usize {
         self.parts
             .values()
-            .filter(|part| part.is_complete(self.threshold))
+            .filter(|part| part.is_complete(self.parts_threshold))
```

Comment:
If I understand this correctly, we actually want total participation here, that is, if we have N participants, we should have N acks for a part to be completed.

I think this is what you end up with because of how you calculate the `parts_threshold`.

```
     n_acks > parts_threshold * 2  
<==> n_acks > ((pub_keys.len() - 1) / 2) * 2
<==> n_acks > pub_keys.len() - 1
<==> n_acks == pub_keys.len()
```

How about we change the logic of `is_complete()` to simply check if we have acks from all participants directly instead of doing this `parts_threshold * 2 + 1` check?

Reply:
Although we currently use it with total participation in `sn`, I don't think we need to enforce such constraints at the `sdkg` module level. The protocol works well from _t * 2 + 1_ on, we use it with total participation which is within the constraint, but we could use it with less someday if we decide that inactive DKG participants can be _outcasted_ from the final keygen. That could be very useful when the network suffers heavy losses and we need to do DKG with fewer nodes.